sympy/physics/mechanics/functions.py

```diff
@@ -532,7 +539,7 @@ def _recurser(expr, sub_dict):
 
 def _fraction_decomp(expr):
     """Return num, den such that expr = num/den"""
-    if not isinstance(expr, Mul):
+    if not isinstance(expr, MulClass):
```

Comment:
But the problem is that it will complicate the wrappers. I am not convinced we should go that way. Because there will be more subtle differences.

I would rather write a compatibility layer for sympy, so that we have truly equivalent interface, on top of the symengine.py.